pkg/generated/models/log_entry.go

```diff
@@ -91,9 +91,9 @@ type LogEntryAnon struct {
 	// attestation
 	Attestation *LogEntryAnonAttestation `json:"attestation,omitempty"`
 
-	// body
+	// The base64-encoded proposed log entry
 	// Required: true
-	Body interface{} `json:"body"`
+	Body *string `json:"body"`
```

Comment:
Hm, this would be a breaking change in terms of the library as we are leveraging the generated models in Cosign (and I assume other Go implementations) - https://github.com/sigstore/cosign/blob/02b1b2677b8fe008ccfea4bbd6f5cc96961a49a8/pkg/cosign/bundle/rekor.go#L42, https://github.com/sigstore/cosign/blob/02b1b2677b8fe008ccfea4bbd6f5cc96961a49a8/pkg/cosign/tlog.go#L256

@bobcallaway Would you be OK with this as a 1.x release, even if this is not necessarily following semver? The other option is waiting to a 2.x release and cleaning this up as part of larger API changes.

Reply:
I'd rather defer to 2.x. Sorry for the extra work @woodruffw :/

Reply:
Let's file a bug to track.

@woodruffw, should we instead continue to omit a format string and simply document that body should be a base64-encoded string? Or would the format hint (even if not spec compliant) be helpful)

Reply:
Documentation is sufficient, I think -- at the very least, the Rust bindings can hack around this (this will be another hack on the stack, but the stack is already quite large 🙂).

That being said: in the context of semver, I've argued before that bugfixes aren't semver violations, and I think there's a good argument that this is purely a bugfix (even through it results in a change to a public API): the current OpenAPI definition for `LogEntry` is incorrect, and results in broken bindings for any downstream consumer that doesn't have the equivalent of Go's `interface` type. Moreover, Go only incidentally works here because API users have to know to punch through the `interface` to get the actual `string` beneath it. So: if Rekor's semver contract includes bugfixes on the OpenAPI definition itself, I _think_ there's a case for this not being a breaking change.

(I can understand if you'd still prefer to defer to 2.x, but I figured I'd advance this argument in case it changes your views here 🙂)

Reply:
Is it a bug to specify an object without any properties? https://stackoverflow.com/questions/67514763/is-it-valid-to-define-an-object-without-any-sub-properties-in-openapi-2-0-swagg suggests this is acceptable (I couldn't find an answer in the spec).

Reply:
> Is it a bug to specify an object without any properties?

I believe that an object without any properties isn't a bug, but that specifying `type: object` when the API really takes `type: string` arguably *is* a bug (since it results in an incorrectly generated client, just in a way that is incidentally fine with Go's `interface`). So my argument is that changing to `type: string` is a bugfix in the semver sense, but this may be too pedantic/practically irrelevant if enough external things use these APIs 😅 

(I think another solution here would be to use `oneOf` and essentially do `type: string | object`. This should retain the existing `interface` type on the Go side but allow for a correct variant in other languages. But this is arguably a bad hack and not worth it if this'll be fixed in 2.0 anyways.)

Reply:
I'm not sure why we originally chose object over string, I'd guess we hadn't standardized on how the API would return `body`. I get what you're saying (and I don't really mind breaking library users, but that's my bias), but I think the impact to existing Go clients will be significant enough that punting on this til v2 would be better.

Hm, `oneOf` is an interesting idea. Does that result in any library changes for the generated code?

Reply:
> Hm, oneOf is an interesting idea. Does that result in any library changes for the generated code?

I'll give that a try and find out!

Reply:
Okay, turns out this is a non-starter: the current OpenAPI spec is written in Swagger 2.0, which doesn't support `oneOf`. So I can't even switch to `oneOf`, unless we pushed the `LogEntry` definition into its own standalone JSON schema file.

Given that, I think I've exhausted all creative alternatives here and this just needs to be fixed in v2 😅. I'll close this and file an associated issue.

Reply:
Do you want to still specify the `format:base64` string as a hint?

Reply:
Doing so doesn't change the hacks we need to do on the Rust side, so I'm fine it omitting it 🙂 -- omitting it arguably also makes the OpenAPI spec slightly more compliant (which isn't a problem at the moment, but could cause problems for an even stricter generator).